libs/libarchfpga/src/read_fpga_interchange_arch.cpp

```diff
@@ -2176,76 +2251,92 @@ struct ArchReader {
         auto tile_types = ar_.getTileTypeList();
         auto site_types = ar_.getSiteTypeList();
 
-        std::vector<std::string> packages;
-        for (auto package : ar_.getPackages())
-            packages.push_back(str(package.getName()));
-
-        for (auto name : packages) {
-            t_grid_def grid_def;
-            grid_def.width = grid_def.height = 0;
-            for (auto tile : tiles) {
-                grid_def.width = std::max(grid_def.width, tile.getCol() + 1);
-                grid_def.height = std::max(grid_def.height, tile.getRow() + 1);
-            }
+        std::string device_name = ar_.getName().cStr();
+        for (auto package : ar_.getPackages()) {
+            for (auto grade : package.getGrades()) {
```

Comment:
It seems completely unnecessary to me to generate what's essentially the same grid layout, just with a different name.
How about generating it once and changing `t_arch::grid_layouts` from `std::vector<t_grid_def>` to `std::unordered_map<std::shared_ptr<t_grid_def>>` so multiple names could point to the same layout?